renderdoc/driver/gl/wrappers/gl_emulated.cpp

```diff
@@ -3036,19 +3047,27 @@ void APIENTRY _glGetTexImage(GLenum target, GLint level, const GLenum format, co
 
     GL.glUseProgram(prog);
 
-    loc = GL.glGetUniformLocation(prog, "srcTex");
-    GL.glUniform1i(loc, 0);
-    loc = GL.glGetUniformLocation(prog, "res");
-    GL.glUniform2f(loc, float(width), float(height));
+    for(int32_t d = 0; d < depth; ++d)
+    {
+      if(target != eGL_TEXTURE_2D)
+      {
+        GL.glFramebufferTextureLayer(eGL_FRAMEBUFFER, eGL_COLOR_ATTACHMENT0, readtex, 0, d);
+      }
 
-    GL.glDrawArrays(eGL_TRIANGLES, 0, 3);
+      loc = GL.glGetUniformLocation(prog, "srcTex");
+      GL.glUniform1i(loc, 0);
+      loc = GL.glGetUniformLocation(prog, "res");
+      GL.glUniform3f(loc, float(width), float(height), float(d));
 
-    // if we support reading stencil, read the stencil into green
-    if(remapformat == eGL_DEPTH_STENCIL && HasExt[ARB_stencil_texturing])
-    {
-      GL.glTexParameteri(target, eGL_DEPTH_STENCIL_TEXTURE_MODE, eGL_STENCIL_INDEX);
-      GL.glColorMask(GL_FALSE, GL_TRUE, GL_FALSE, GL_FALSE);
       GL.glDrawArrays(eGL_TRIANGLES, 0, 3);
+
+      // if we support reading stencil, read the stencil into green
+      if(remapformat == eGL_DEPTH_STENCIL && HasExt[ARB_stencil_texturing])
+      {
+        GL.glTexParameteri(target, eGL_DEPTH_STENCIL_TEXTURE_MODE, eGL_STENCIL_INDEX);
+        GL.glColorMask(GL_FALSE, GL_TRUE, GL_FALSE, GL_FALSE);
+        GL.glDrawArrays(eGL_TRIANGLES, 0, 3);
+      }
```

Comment:
For the 2D Array/3D case you should re-set the depth-stencil texture mode back to `eGL_DEPTH_COMPONENT` after each iteration, so that the depth draw on the next slice will be correct. Otherwise an arrayed depth-stencil texture will read stencil for both parts of slices 1..n.

Reply:
Thank you for quick and very nice review. Yes, in new version I re-set depth component and also colorMask too.
Do I need to squash commits by myself, or you can do it?

Reply:
I only rebase branches when merging I don't ever squash, so yes please squash and force-push on your branch.

Reply:
I squashed it.